gradio/routes.py

```diff
@@ -547,44 +547,22 @@ async def file(path_or_url: str, request: fastapi.Request):
                 raise HTTPException(403, f"File not allowed: {path_or_url}.")
 
             abs_path = utils.abspath(path_or_url)
-
-            in_blocklist = any(
-                utils.is_in_or_equal(abs_path, blocked_path)
-                for blocked_path in blocks.blocked_paths
-            )
-
-            is_dir = abs_path.is_dir()
-
-            if is_dir or in_blocklist:
+            if abs_path.is_dir() or not abs_path.exists():
                 raise HTTPException(403, f"File not allowed: {path_or_url}.")
```

Comment:
nit: it would be better to set an `allowed` flag here, and raise a single 403 later, after the `utils.is_allowed_file` has completed so that malicious attackers cannot using timing attacks to infer whether a particular filepath even exists (that itself can be insecure in certain settings)

Reply:
Yea good call.